File: master_the_coding_interview_course/recursion/factorial.py
#!/usr/bin/env python3

def findFactorialRecursive(number):
    if number == 1:
        return number
    return number * findFactorialRecursive(number - 1)



print(findFactorialRecursive(5))

def findFactorialIterative(number):
    factorial = 1
    while number > 1:
        factorial *= number
        number -= 1
    return factorial


print(findFactorialIterative(5))

File: master_the_coding_interview_course/recursion/factorial.js
function findFactorialRecursive(num){
    if(num === 1){
        console.log(num);
        return num
    }
    console.log(num);
    let factorial = findFactorialRecursive(num - 1) * num;
    console.log(factorial);
    return factorial
}

console.log(findFactorialRecursive(10));

function findFactorialIterative(num){
    let factorial = 1;
    while(num > 0){
        factorial *= num;
        num--;
    }
    return factorial;
}

console.log(findFactorialIterative(5));